src/service/url-replacements-impl.js

```diff
@@ -82,6 +82,9 @@ export class GlobalVariableSource extends VariableSource {
 
     /** @private {?Promise<?ShareTrackingFragmentsDef>} */
     this.shareTrackingFragments_ = null;
+
+    /** @private {?Promise<?{pageIndex: number, pageId: string}>} */
```

Comment:
This should return `AmpStoryVariableService` right?

Reply:
That would be ideal, but I'd rather not add the whole `amp-story` tree to the [general source file set](https://github.com/ampproject/amphtml/blob/master/build-system/tasks/compile.js#L126) (otherwise the module can't be resolved).

Reply:
We're not really retrieving an element "service" here though. How are we registering that data as a service with the runtime? `registerService`?

Reply:
[`registerServiceBuilder`](https://github.com/newmuis/amphtml-story/blob/master/extensions/amp-story/0.1/amp-story.js#L150)

Reply:
Ohh I see. 😃  I think adding the extension to the Closure file path is fine. Erwin's working on making Closure compile runtime + extensions in a single pass anyways. Replacing a class with a record-type of its getter property interface confused me a lot.

Reply:
This is super confusing. Why is it a record? Don't these change? There are a lot of services here that function exactly same way, e.g. `getAccessService_`.

This definitely needs to change. Please send a PR.